resolwe_bio/tests/processes/test_diff_expression.py

```diff
@@ -163,3 +163,111 @@ def test_edger(self):
+                    },
+                },
+                data_name="shRNA expression ({{ reads|sample_name|default('?') }})",
+                # data_name="shRNA expression ({{ input_data | name | default('?') }})",
+                contributor=self.contributor,
+                type='data:expression:bowtie2quant:',
+                input_schema=[
+                    {
+                        'name': 'exp',
+                        'type': 'basic:file:',
+                    }
+                ],
+                output_schema=[
+                    {
+                        'name': 'exp',
+                        'type': 'basic:file:',
+                    },
+                    {
+                        'name': 'rc',
+                        'type': 'basic:file:'
+                    },
+                    {
+                        'name': 'exp_type',
+                        'type': 'basic:string:'
+                    },
+                    {
+                        'name': 'source',
+                        'type': 'basic:string:'
+                    },
+                    {
+                        'name': 'species',
+                        'type': 'basic:string:'
+                    },
+                    {
+                        'name': 'build',
+                        'type': 'basic:string:'
+                    },
+                    {
+                        'name': 'feature_type',
+                        'type': 'basic:string:'
+                    },
+                    {
+                        'name': 'mapped_species',
+                        'type': 'basic:file:'
+                    }
+                ],
+                run={
+                    'language': 'bash',
+                    'program': r"""
+            re-import {{ exp.file_temp|default(exp.file) }} {{ exp.file }} "txt" "txt" 0.1 compress
+            
+            cp "${NAME}".txt.gz "${NAME}"_trimmed_trimmed_count_matrix.txt.gz
+            
+            # re-save-file exp "${NAME}".txt.gz
+            # re-save-file rc "${NAME}".txt.gz
+            re-save-file exp "${NAME}"_trimmed_trimmed_count_matrix.txt.gz
+            re-save-file rc "${NAME}"_trimmed_trimmed_count_matrix.txt.gz
+            re-save exp_type 'RC'
+            re-save source 'shRNA-gene-sequences'
+            re-save species 'Homo sapiens'
+            re-save build 'custom-from-file'
+            re-save feature_type 'shRNA'
+            # Uploading fake mapped species file, do not use in tests.
+            re-save-file mapped_species "${NAME}".txt.gz
+            """
+                }
+            )
+
+            sample_filenames = [pf_in + f'sample-{i}.txt.gz' for i in list(range(1, 13))]
+
+            list_expressions = []
+            for i in sample_filenames:
+                list_expressions.append(self.run_process(process.slug, {'exp': i}))
+
+        inputs = {
+            'parameter_file': pf_in + parameter_file.output['file']['file'],
```

Comment:
If `differentialexpression-shrna` is re-worked to accept the DataField on inputs, you can write just `'parameter_file': parameter_file.id` here.

Reply:
Done.